starknetdata/feeder/feeder.go

```diff
@@ -113,3 +113,35 @@ func (f *Feeder) StateUpdate(ctx context.Context, blockNumber uint64) (*core.Sta
 func (f *Feeder) StateUpdatePending(ctx context.Context) (*core.StateUpdate, error) {
 	return f.stateUpdate(ctx, "pending")
 }
+
+func (f *Feeder) stateUpdateWithBlock(ctx context.Context, blockID string) (*core.StateUpdate, *core.Block, error) {
```

Comment:
you can move the fallback logic to here. It doesnt need to be in `sync` package

Reply:
I avoided that to not be forced to add the fallbacks to the `StarknetData` interface, and hence inject them in mocks too.

Reply:
This fallback mechanism is a feeder specific detail that we shouldnt leak in to `sync` package. 

> hence inject them in mocks too

only `StarknetData` interface affects the mocks. The fallback that you implement in `Feeder` has no affect on mocks.

Reply:
True, thanks